api/ipam/v1beta2/ipaddress_types.go

```diff
@@ -48,6 +47,43 @@ type IPAddressSpec struct {
 	Gateway string `json:"gateway,omitempty"`
 }
 
+// IPAddressClaimReference is a reference to an IPAddressClaim.
+type IPAddressClaimReference struct {
```

Comment:
I'll review the details here once the CC PR Is final and this one "rebased" on top of it